src/main/java/org/springframework/data/neo4j/core/mapping/DefaultNeo4jPersistentProperty.java

```diff
@@ -182,7 +183,7 @@ public boolean isAssociation() {
 
 	@Override
 	public boolean isEntity() {
-		return super.isEntity() && isAssociation() || (super.isEntity() && isEntityInRelationshipWithProperties());
+		return super.isEntity() && isAssociation() || (super.isEntity() && isEntityInRelationshipWithProperties() && !isAnnotationPresent(CompositeProperty.class));
```

Comment:
```suggestion
		return super.isEntity() && isAssociation() || (super.isEntity() && isEntityInRelationshipWithProperties() && !isComposite());
```

Reply:
I prefer a central access to the annotation more than having the same (inversed) logic written again.

Reply:
Me too. Thanks.